### DISCUSSION TITLE
Feature request: Dynamic Onboarding PR adjustments

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#21666] [Suggest an Idea]
Right now the generation of Onboarding PR is pretty "static", but there would be some nice to have things we could potentially do instead.

For example [remove pr hourly limits if we detect buildkite](https://github.com/renovatebot/config-help/issues/62#issuecomment-403182250).

Or: warn the user if we detected that their base branch (e.g. `master`) is failing tests already.

For self-hosted bot owners, maybe they even want to write their own JS function to perform this customisation.

**Comment by hutson** [comment]
At my company we currently use the [`onboardingConfig`](https://renovatebot.com/docs/self-hosted-configuration/#onboardingconfig) to have the initial on-boarding PR use our `web` preset as the default configuration. For the past 6 months, it's worked quite well. Going forward we're looking to begin on-boarding other languages, product groups, developer communities, each with their own subtlety different needs. We would like dynamically generate our configuration (likely just select an existing, context-specific, preset) based on repository content, repository name, SCM platform, etc.

**Comment by rarkins** [comment]
@hbetts I wasn’t expecting that you would be able to make use of this feature though if you are restricted to a 12-factor env approach?

**Comment by hutson** [comment]
> @hbetts I wasn’t expecting that you would be able to make use of this feature though if you are restricted to a 12-factor env approach?

Ugh.... yeah. The best we have is environment variables (Assuming we don't fall back on deploying a custom version of Renovate with what we need).
